libs/designer-ui/src/lib/picker/filepickereditor.tsx

```diff
@@ -38,35 +38,67 @@ const calloutProps = { gapSpace: 0 };
 export const FilePickerEditor = ({
   initialValue,
   isLoading = false,
+  type,
+  items,
+  displayValue,
+  // fileFilters,
+  errorDetails,
   editorBlur,
-  onChange,
   pickerCallbacks,
   ...baseEditorProps
 }: FilePickerEditorProps) => {
-  const [value, setValue] = useState(initialValue);
-  const [showPicker, setShowPicker] = useState(false);
-  const [titleSegments] = useState<PickerTitleInfo[] | undefined>();
   const pickerIconId = useId();
   const intl = useIntl();
+  const [selectedItem, setSelectedItem] = useState<any>();
+  const [editorDisplayValue, setEditorDisplayValue] = useState<ValueSegment[]>(
+    displayValue ? [{ id: guid(), value: displayValue, type: ValueSegmentType.LITERAL }] : initialValue
+  );
+  const [pickerDisplayValue, setPickerDisplayValue] = useState<ValueSegment[]>([]);
+  const [showPicker, setShowPicker] = useState(false);
+
+  useEffect(() => {
+    if (displayValue) {
+      setPickerDisplayValue([{ id: guid(), value: displayValue, type: ValueSegmentType.LITERAL }]);
+    }
+  }, [displayValue]);
 
-  const { onFolderNavigation: onFolderNavigated } = pickerCallbacks;
+  const { onFolderNavigation, getFileSourceName, getDisplayValueFromSelectedItem, getValueFromSelectedItem } = pickerCallbacks;
+  const fileSourceName = getFileSourceName?.();
```

Comment:
All callbacks are required so getFileSourceName need not have ? and similar to other callbacks as well in this file.